### PR TITLE
Rotate view for 3d viewers

### PR DESCRIPTION
**Pull request by PennyQ**
I use a `check box` to start the `Timer` that controls the rotation, and a `slider bar` to set the rotation speed. But there are some conflicts between other setting controls like stretch and min/max limit. Should we disable some when we start the rotation or let them work simultaneously? @astrofrog 

image

### CODE REVIEW

**Comment by PennyQ** [comment]
Check failed with the same problem of `MatrixTransform` as in https://github.com/glue-viz/glue-3d-viewer/pull/96

**Comment by astrofrog** [comment]
@PennyQ - instead of modifying the transform, just change the azimuth of the camera by doing:

```
camera.azimuth += 1
```

for example. This will avoid issues with the other options.

**Comment by PennyQ** [comment]
I feel  if we choose to modify the camera to realize the rotation view,  both azimuth and position should be changed to let the camera 'orbit' the object.

**Comment by astrofrog** [comment]
@PennyQ - ah I see - what is currently chosen by default as the 'center of the orbit' if you just change the azimuth?

**Comment by PennyQ** [comment]
@astrofrog Here is the comment from [TurntableCamera ](https://github.com/vispy/vispy/blob/f6c0e03e36ec402f1f3a1fb5909dc09575b4fb69/vispy/scene/cameras/turntable.py) : `3D camera class that orbits around a center point while maintaining a view on a center point.`

With a figure showing the space relation :)
image
Credit: http://www.codinglabs.net/article_world_view_projection_matrix.aspx

**Comment by astrofrog** [comment]
@PennyQ - thanks! I guess what I mean is that we are using turntable camera:

https://github.com/glue-viz/glue-3d-viewer/blob/master/glue_vispy_viewers/common/vispy_widget.py#L54

so what happens if you do try and change azimuth?

**Comment by PennyQ** [comment]
@astrofrog Yeah you are right, we can use the `azimuth` to do the rotate and it's easier :)

File: glue_vispy_viewers/common/vispy_widget.py [hunk 1]
```diff
@@ -73,15 +81,25 @@ def _toggle_axes(self):
...
+            self.timer.stop()
+
     def add_data_visual(self, visual):
         self.limit_transforms[visual] = scene.STTransform()
         visual.transform = self.limit_transforms[visual]
         self.view.add(visual)
 
     def _update_stretch(self, *stretch):
-        self.scene_transform.scale = stretch
+        self.axis_transform.scale = stretch
```
Comment (astrofrog) [inline glue_vispy_viewers/common/vispy_widget.py:96]: Is this change needed?

File: glue_vispy_viewers/common/vispy_widget.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ def __init__(self, parent=None):
         self.axis = scene.visuals.Mesh(vertices['position'],
                                        outline_indices,
                                        color=(1,1,1), mode='lines')
-        self.axis.transform = self.scene_transform
+        self.axis.transform = self.axis_transform
```
Comment (astrofrog) [inline glue_vispy_viewers/common/vispy_widget.py:46]: Is this change needed?

File: glue_vispy_viewers/common/vispy_widget.py [hunk 3]
```diff
@@ -32,7 +33,7 @@ def __init__(self, parent=None):
         self.emulate_texture = (sys.platform == 'win32' and
                                 sys.version_info[0] < 3)
 
-        self.scene_transform = scene.STTransform()
+        self.axis_transform = scene.STTransform()
```
Comment (astrofrog) [inline glue_vispy_viewers/common/vispy_widget.py:36]: Is this change needed?

File: glue_vispy_viewers/common/vispy_widget.py [hunk 4]
```diff
@@ -108,6 +126,9 @@ def _update_limits(self):
     def _reset_view(self):
         self.view.camera.reset()
 
+    def rotate(self, event):
+        self.rotate_speed += self.rotate_slider_val
+        self.view.camera.azimuth = self.rotate_speed
```
Comment (astrofrog) [inline glue_vispy_viewers/common/vispy_widget.py]: Here I wonder whether it wouldn't make more sense to just do:

```
self.view.camera.azimuth += self.rotate_slider_val
```

(otherwise it sounds like the speed is increasing all the time)

File: glue_vispy_viewers/common/vispy_widget.py (hunk 3 above)
Comment (PennyQ) [inline glue_vispy_viewers/common/vispy_widget.py:36]: @astrofrog I found it's a little confusing for this variable name coz this `self.scene_transform` is only used for setting the axis's transform so I changed it's name to make it more clear, I can change it back if we want to keep it :)

Comment (astrofrog) [inline glue_vispy_viewers/common/vispy_widget.py:36]: No that's fine, just checking :)

**Comment by PennyQ** [comment]
for #67

**Comment by astrofrog** [comment]
To keep things simple for now, and to avoid confusion with the play/pause we are suggesting for the _sliders_, I suggest that we simply add a single button to the toolbar that makes it rotate in a fixed direction at a constant speed. Maybe we could use something like this as an icon for now:

http://fortawesome.github.io/Font-Awesome/icon/undo/

@PennyQ - what do you think?

**Comment by PennyQ** [comment]
@astrofrog Yeah sounds good, can you show me how to save this icon to a `.png` file and I could help to add it into the viewer :)

**Comment by PennyQ** [comment]
Create a new PR #128 and close this one :)
